src/xclim/indices/_hydrology.py

```diff
@@ -647,3 +648,43 @@ def low_flow_frequency(q: xarray.DataArray, threshold_factor: float = 0.2, freq:
     threshold = threshold_factor * mean_flow
     out = threshold_count(q, "<", threshold, freq=freq)
     return to_agg_units(out, q, "count")
+
+
+@declare_units(pr="[precipitation]")
+def antecedent_precipitation_index(pr: xarray.DataArray, window: int = 7, p_exp: float = 0.935) -> xarray.DataArray:
+    """
+    Antecedent Precipitation Index.
+
+    Calculate the running weighted sum of daily precipitation values given a window and weighting exponent.
+    This index serves as an indicator for soil moisture.
+
+    Parameters
+    ----------
+    pr : xarray.DataArray
+        Daily precipitation data.
+    window : int, optional
+        Window for the days of precipitation data to be weighted and summed, default is 7.
+    p_exp : float, optional
+        Weighting exponent, default is 0.935.
```

Comment:
```suggestion
    window : int
        Window for the days of precipitation data to be weighted and summed, default is 7.
    p_exp : float
        Weighting exponent, default is 0.935.
```
"optional" usually means that `None` can be given, which is not the case here.